### ISSUE TITLE
Managing message strings (i18n)

### ISSUE THREAD

**Issue by gohanman** [CORE-POS/IS4C#52]
@DonaldGonzalez mentioned something on another issue about fixing capitalization on user-facing messages. I think if we're going through and re-writing/fixing a bunch of strings, it's a good opportunity to do it _correctly_ and use some built-in string management.

I spent a little time getting familiar with PHP's gettext implementation and wrote up some simple documentation:
http://is4c.gohanman.com/git/IS4C/documentation/IS4C/developer/i18n.html

It's pretty easy to use, from a developer standpoint. Basically, put _() around user-facing strings. I doubt we need alternate language versions any time soon, but moving the messages into one file makes some future changes easier. If we want to change in terminology or formatting, there's no need to go hunting through all the code files. Everything's in the en_US translation.

(I marked this low priority because it will be somewhat time consuming; I'm not suggesting the original suggestion is unimportant)

**Comment by joelbrock** [comment]
I'm reopening as this has come up again from another co-op looking to globally manage the terminology used by CORE, e.g. "owners" not "members".

I realize that fully implementing this is a bit of an undertaking but I'd like to make a case for having this on a roadmap for future releases.  Terminology / nomenclature is something that is consistently inconsistent from one co-op to the next so having a robust (or at least complete) implementation using .pot files would be a great step fwd.  

To clarify:  I think our burden would be on applying this to the CORE Office and Lane core code.  Plugins and customized modules would be on the authors to update, IMO.

**Comment by gohanman** [comment]
I don't think the task is daunting some much as tedious to make sure every user facing string gets wrapped in a `_()` function call.

One other piece that I think is worth noting in pursuing this is that it's not a good idea to wrap individual words but rather whole phrases. So the custom translation might have to replace "member" a bunch of times in different phrases. e.g.,
```php
        } elseif (!$this->submitted) {
            echo _("member search")."<br />"._("enter member number or name");
        } else {
            echo _("no match found")."<br />"._("next search or member number");
        }
```

It's a moderate annoyance for the use case we're talking about, although find and replace can probably handle much of it. But if someone wanted to translate the UI to **not English**, then having translations at the level of individual words doesn't work at all. If we're going to put the effort into wrapping all these strings it might as well be in a way that maximizes translatability.

**Comment by gohanman** [comment]
Update: the lane side should be essentially done. [This doc](https://github.com/CORE-POS/IS4C/wiki/Internationalization) appears to still be accurate, which was handy since I'd forgotten all the commands.

Another piece of this relates to collaboration. Upstream currently provides en_US and en_CA. I'd like to treat en_CA as more-or-less belonging to @flathat until another relevant user comes along. For en_US, I think any changes or fixes should happen directly in PHP rather than in the .po/.mo files. That will make other translations files easier to deal with since `msgid`s match the default output.

For "local custom terminology translations" that are still English, downstream users **should not** use en_US or en_CA. That will avoid merge conflict problems. In theory anything else could be used but on most systems there are plenty of en_* variants to choose from. Future upstream translations (if any) will be published as standalone repos so "official" translations don't create merge conflicts with local custom translations.

**Comment by gohanman** [comment]
This one (db4f11881e9dfa9409901e03d6646bc1281dc001) merits further discussion.

I started down the same road with Office and on literally the first file realized that concatenating `_()` calls on the backend is going to be painful and tedious. So I installed [twig](http://twig.sensiolabs.org/). It only affects the one [non-vital] admin index page at the moment, but this would officially make composer required. That page will not work without installing the twig packages.

This would remain opt-in. Just because a page has a `$twig` member doesn't mean you're obligated to use it. But using this approach for i18n probably means that more and more backend pages will make use of twig templates and other people who work with those codebase will eventually bump into these templates.
